tests/MongoDB.Driver.Tests/Linq/MongoQueryableWithDotNotationTests.cs

```diff
@@ -26,62 +27,82 @@ public class MongoQueryableWithDotNotationTests
 
-            result.ToString().Should().Be("aggregate([{ \"$match\" : { \"ExtraInfo.Type\" : null } }])");
+                result.ToString().Should().Be("aggregate([{ \"$match\" : { \"ExtraInfo.Type\" : null } }])");
+            }
         }
 
         [Fact]
         public void Where_with_ExtraInfo_Type_with_Value_should_render_correctly()
         {
-            var subject = CreateSubject();
+            using (var client = CreateDisposableClient())
+            {
+                var subject = CreateSubject(client);
 
-            var result = subject.Where(c => c.ExtraInfo.Type.Value == 2);
+                var result = subject.Where(c => c.ExtraInfo.Type.Value == 2);
 
-            result.ToString().Should().Be("aggregate([{ \"$match\" : { \"ExtraInfo.Type\" : 2 } }])");
+                result.ToString().Should().Be("aggregate([{ \"$match\" : { \"ExtraInfo.Type\" : 2 } }])");
+            }
         }
 
         [Fact]
         public void Where_with_ExtraInfo_Type_with_Value_and_nullable_variable_should_render_correctly()
         {
-            var subject = CreateSubject();
-            int? infoType = 3;
+            using (var client = CreateDisposableClient())
+            {
+                var subject = CreateSubject(client);
+                int? infoType = 3;
 
-            var result = subject.Where(c => c.ExtraInfo.Type.Value == infoType);
+                var result = subject.Where(c => c.ExtraInfo.Type.Value == infoType);
 
-            result.ToString().Should().Be("aggregate([{ \"$match\" : { \"ExtraInfo.Type\" : 3 } }])");
+                result.ToString().Should().Be("aggregate([{ \"$match\" : { \"ExtraInfo.Type\" : 3 } }])");
+            }
         }
 
         [Fact]
         public void Where_with_Contains_should_render_correctly()
         {
-            var subject = CreateSubject();
-            var list = new List<int>
+            using (var client = CreateDisposableClient())
             {
-                4, 5
-            };
+                var subject = CreateSubject(client);
+                var list = new List<int>
+                {
+                    4, 5
+                };
 
-            var result = subject.Where(c => list.Contains(c.ExtraInfo.Type.Value) || list.Contains(c.ExtraInfo.NotNullableType));
+                var result = subject.Where(c => list.Contains(c.ExtraInfo.Type.Value) || list.Contains(c.ExtraInfo.NotNullableType));
 
-            result.ToString().Should().Be("aggregate([{ \"$match\" : { \"$or\" : [{ \"ExtraInfo.Type\" : { \"$in\" : [4, 5] } }, { \"ExtraInfo.NotNullableType\" : { \"$in\" : [4, 5] } }] } }])");
+                result.ToString().Should().Be("aggregate([{ \"$match\" : { \"$or\" : [{ \"ExtraInfo.Type\" : { \"$in\" : [4, 5] } }, { \"ExtraInfo.NotNullableType\" : { \"$in\" : [4, 5] } }] } }])");
+            }
         }
 
         // private methods
-        private IQueryable<Car> CreateSubject()
+        private DisposableMongoClient CreateDisposableClient()
```

Comment:
I would check all our tests to ensure that we don't use MongoClient without disposing resourses

Reply:
NOTE: I noticed this particular test file in my debug steps a lot, because of different connection string

Reply:
>we could potentially create a separate task for the sync/async versions of the integration tests if it'll help us temporarily bypass server issues until they are resolved

for now, it looks like we don't have this issue anymore (*finger cross*) :)